services/ui-src/src/index.tsx

```diff
@@ -22,17 +24,19 @@ Amplify.configure({
 });
 
 ReactDOM.render(
-  <Router>
-    <UserProvider>
-      <ApiProvider>
-        <QueryProvider>
-          <ChakraProvider theme={theme}>
-            <App />
-          </ChakraProvider>
-          {/* <ReactQueryDevtools /> */}
-        </QueryProvider>
-      </ApiProvider>
-    </UserProvider>
-  </Router>,
+  <ErrorBoundary FallbackComponent={Error}>
+    <Router>
+      <UserProvider>
+        <ApiProvider>
+          <QueryProvider>
+            <ChakraProvider theme={theme}>
+              <App />
+            </ChakraProvider>
+            {/* <ReactQueryDevtools /> */}
+          </QueryProvider>
+        </ApiProvider>
+      </UserProvider>
+    </Router>
+  </ErrorBoundary>,
```

Comment:
Wrapping the whole thing in an error boundary at the root just in case (best practice).